rotkehlchen/tests/api/blockchain/test_evm.py

```diff
@@ -284,3 +289,48 @@ def test_add_multievm_accounts(rotkehlchen_api_server):
     assert result == [
         {'address': common_account, 'label': label, 'tags': ['metamask']},
     ]
+
+
+@pytest.mark.parametrize('ethereum_accounts', [[ADDY]])
+def test_evm_account_deletion_does_not_wait_for_pending_txn_queries(rotkehlchen_api_server) -> None:  # noqa: E501
+    """
+    Test that if transactions for an address is being queried and account removal is issued,
+    the transactions querying greenlet is killed & the account is deleted subsequently.
+    """
+    # make an async query of all transactions.
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'evmtransactionsresource',
+        ), json={
+            'async_query': True,
+            'only_cache': False,
+            'limit': 1000,
+            'accounts': [{'address': ADDY, 'evm_chain': 'ethereum'}],
+            'evm_chain': 'ethereum',
+        },
+    )
+    assert_ok_async_response(response)
```

Comment:
Don't let this test make remote queries. Since we want to emulate a long running transaction query patch one of the underlying functions of the transaction query (not the one the greenlet calls), to run an infinite loop (with gevent sleeps in between so it yields)